src/main/java/com/onto/javacourse/intro/findingArraysMedian/FindingArraysMedian.java

```diff
@@ -0,0 +1,87 @@
+package com.onto.javacourse.intro.findingArraysMedian;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class FindingArraysMedian
+{
+    private static final Logger log = LoggerFactory.getLogger(FindingArraysMedian .class);
+
+    /**
+     * @return Calculates the 1-based position of the median (The sum of the elements before and after the median should be as close as possible. The
+     * median is an element from the array.) in the given array.
+     * @param array the array of integers to find the median of
+     * @author Kristian Vasilev
+     */
+    public int median(int[] array) {
+        if (array.length == 1) {
```

Comment:
Imo this is not a good approach. Basically in the method that should do the job you have only implemented validation of the input. Then I have to scroll to the private method to get the actual code. Can't you do it the other way around? For example check input and throw error or return -1, whatever.